src/threading/thread.cpp

```diff
@@ -156,21 +155,30 @@ bool Thread::stop()
 }
 
 
-void Thread::wait()
+bool Thread::wait()
 {
-	if (!m_running)
-		return;
+	MutexAutoLock lock(m_mutex);
+
+	if (!m_joinable)
```

Comment:
That's why the lock is there. once the thread is joined, nobody else can obtain the lock. So a check for m_running is sufficient, no?